python/cudf/cudf/tests/test_concat.py

```diff
@@ -237,3 +237,33 @@ def test_concat_string_index_name(myindex):
     df3 = gd.concat([df1, df2])
 
     assert df3.index.name == myindex
+
+
+def test_pandas_concat_compatibility_axis1():
```

Comment:
Could you add an additional test with some overlap in the indexes? The below will likely almost never have overlap.